### PR TITLE
Enabling ColBERT CI test on a GPU

### LINKED ISSUE #451: pytest tests/primeqa/ir/dense/colbert_top/colbert/test_training.py fails on a single GPU machine
When I run this test on a single GPU machine data parallel distributed training is picked up (looks like the test does not gracefully fall back to a single GPU option). As a result, the pytorch errors out with - 

Default process group has not been initialized.

### PR DESCRIPTION
**Pull request by franzmpub**
# Pull Request      

## What does this PR do?

Closes #451 
 
## Description

The `setup_class` function in the `TestTraining` class contains code to call `torch.distributed.init_process_group` when a GPU is available.

## Versioning

When opening a PR to make changes to PrimeQA (i.e. `primeqa/`) master, be sure to increment the version following
[semantic versioning](https://semver.org/).  The VERSION is stored [here](https://github.com/primeqa/primeqa/blob/main/VERSION) 
and is incremented using `bump2version {patch,minor,major}` as described in the (development guide documentation)[development.html].

 - [x] Have you updated the VERSION?
 - [ ] Or does this PR not change the `primeqa` package or was not into master?

After pulling in changes from master to an existing PR, ensure the VERSION is updated appropriately.
This may require bumping the version again if it has been previously bumped.

If you're not quite ready yet to post a PR for review, feel free to open a [draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/).

### After Merging
If merging into master and VERSION was updated, after this PR is merged:
- [ ] [Create a release](https://docs.github.com/en/github/administering-a-repository/releasing-projects-on-github/managing-releases-in-a-repository#creating-a-release) from the master with version equal to [VERSION](https://github.com/primeqa/primeqa/blob/main/VERSION)
- [ ] Not merging to master or VERSION not updated

### Checklist

Review the following and mark as completed:
- [x] [Tag an issue or issues](#what-does-this-pr-do) this PR addresses.
- [x] Added [description](#description) of changes proposed.
- [x] Review requested as [appropriate](#request-review).
- [x] Version bumped as [appropriate](#Versioning).
- [x] New classes, methods, and functions documented.
- [ ] Documentation for modified code is updated.
- [ ] Built documentation to confirm it renders as expected (see [here](development.md)).
- [ ] Code cleaned up and commented out code removed.
- [ ] Tests added to ensure all functionalities tested at >= 60% unit test coverage (see [here](development.md)).
- [ ] Code cleaned up and commented out code removed.
- [ ] Release created [as needed](#after-merging) after merging.

### CODE REVIEW

**Review by bhavani105 — APPROVED** [review]
LGTM

**Review by hrojek — APPROVED** [review]
LGTM

File: tests/primeqa/ir/dense/colbert_top/colbert/test_training.py [hunk 1]
```diff
@@ -17,6 +18,29 @@
...
 
 class TestTraining(UnitTest):
+    @classmethod
+    def setup_class(cls):
+        import torch
+
+        if torch.cuda.is_available():
+            import time
+            import random
+            import os
```
Comment (seirasto) [inline tests/primeqa/ir/dense/colbert_top/colbert/test_training.py:28]: This is already imported at the top of the file

**Review by seirasto — APPROVED** [review]
LGTM

**Pull request merged**